core/base/inc/TDirectory.h

```diff
@@ -262,10 +262,19 @@ can be replaced with the simpler and exception safe:
    /// \param[in] name Name of the object in the file.
    /// \param[in] option Options. See TDirectory::WriteTObject() or TDirectoryWriteObjectAny().
    /// \param[in] bufsize Buffer size. See TDirectory::WriteTObject().
-   template <class T> inline Int_t WriteObject(const T* obj, const char* name, Option_t *option="", Int_t bufsize=0)
-      {
-         return WriteObjectAny(obj, TClass::GetClass<T>(), name, option, bufsize);
-      }
+   template <typename T>
+   inline typename std::enable_if_t<!std::is_base_of<TObject, T>::value, Int_t>
```

Comment:
is the `typename` really needed, here and below? https://godbolt.org/z/zEfxcWhbe

Reply:
Indeed, thanks a lot for the hint! :smile: